.clang-format

```diff
@@ -0,0 +1,60 @@
+BasedOnStyle: LLVM
```

Comment:
I think it worth to discuss the rules and work out a final code style.

Clang-format is not perfect, but it is good enough to keep the code style consistent. I am open to different options for code styles. Sometimes our 'feeling' is not consistent, thus it's not as reliable as rules.